### PR TITLE
Add registrar endpoint support

### PR DESCRIPTION
**Pull request by jacegu**
This PR adds support for the endpoints described on https://developer.dnsimple.com/v2/registrar/

### CODE REVIEW

**Comment by jacegu** [comment]
I had to rewrite part of the fixture parsing code because it was failing with several fixtures and I couldn't make it work. Still working on supporting all the registrar behaviour.

**Comment by weppos** [comment]
Which issue were you expecting? FYI, some fixtures are "invalid", in the sense that they have `\n` as line feed, whereas it should be `\r\n`. They were "changed" (probably by vim) when I edited it.

I've fixed most of them, but there's still some.
Example https://github.com/aetrion/dnsimple-developer/commit/1f9cf5a83677b2e59b0125d081fb124d97898274

I had some troubles with Go with invalid fixtures. If that's the issue, I'd rather fix the fixtures once for all. Ruby is very permissive, and I never noticed such formatting errors. Also in Go I had a problem with the "transfer chunked" header, I had to strip it when reading the fixture.

Not sure if it helps.

File: lib/dnsimple/registrar_service.ex [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  def check_domain(client, account_id, domain_name, headers \\ [], options \\ []) do
+    url = Client.versioned("/a/#{account_id}/registrar/domains/#{domain_name}/check")
+
+    Client.get(client, url, options)
+      |> Response.parse(DomainCheck)
+  end
+
+  @spec register_domain(Client.t, String.t, String.t, Keyword.t, Keyword.t, Keyword.t) :: Response.t
+  def register_domain(client, account_id, domain_name, attributes \\ [], headers \\ [], options \\ []) do
+    url = Client.versioned("/a/#{account_id}/registrar/domains/#{domain_name}/registration")
```
Comment (weppos) [inline lib/dnsimple/registrar_service.ex]: FYI, the path of the API is wrong. There is no `/a`. I also suggest to try it live at least once to make sure it works.

In the Go and Ruby client I added a "live" suite, that tests some of the common API against production. At some point, I was planning to add it here too.

**Comment by jacegu** [comment]
@weppos this is pretty much done.

This branch adds support for other (wrong) delimiters in `test/exvcr_utils.ex`. We can keep it as is and fix the fixtures outside the scope of the branch. I added some tests so we can modify it more easily.

**Comment by jacegu** [comment]
@weppos I am going to keep working on other endpoints branching off from this one so I can get the improvements added to `test/exvcr_utils.ex` there too.

**Comment by weppos** [comment]
@jacegu it looks good to me, great work!

If you want, I'm fine to merge this PR and move forward.

File: lib/dnsimple/registrar_service.ex [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  end
+
+  @doc """
+  Renews a domain.
+
+  See https://developer.dnsimple.com/v2/registrar/#renew
+  """
+  @spec renew_domain(Client.t, String.t, String.t, Keyword.t, Keyword.t, Keyword.t) :: Response.t
+  def renew_domain(client, account_id, domain_name, attributes \\ [], headers \\ [], options \\ []) do
+    url = Client.versioned("/#{account_id}/registrar/domains/#{domain_name}/renew")
```
Comment (weppos) [inline lib/dnsimple/registrar_service.ex]: I believe the path is wrong.
See https://developer.dnsimple.com/v2/registrar/#renew

Comment (jacegu) [inline lib/dnsimple/registrar_service.ex]: Good catch. It is wrong on the Ruby client too. Fixing that.

Thanks!

**Pull request merged**